modules/node-app-service/provider.tf

```diff
@@ -2,8 +2,14 @@ terraform {
   required_providers {
     azurerm = {
       source                = "hashicorp/azurerm"
-      version               = "~> 3.0"
+      version               = ">= 3.0, < 5.0"
       configuration_aliases = [azurerm, azurerm.tooling]
     }
   }
+  required_version = ">= 1.5.7, < 1.9.6"
+}
+
+provider "azurerm" {

```

Comment:
Without this here, validation will fail. However, I believe that when the other repos calling this module their validation will fail with this in as the root module will try to pass its own config, a child module should not have its own configuration.

`❯ terraform validate
╷
│ Error: Provider configuration not present
│
│ To work with data.azurerm_container_registry.acr its original provider configuration at provider["registry.terraform.io/hashicorp/azurerm"].tooling is required, but it has been removed. This occurs     
│ when a provider configuration is removed while objects created by that provider still exist in the state. Re-add the provider configuration to destroy data.azurerm_container_registry.acr, after which   
│ you can remove the provider configuration again.`